pytest_trio/plugin.py

```diff
@@ -215,11 +221,25 @@ def _install_async_fixture_if_needed(fixturedef, request):
 @pytest.hookimpl(hookwrapper=True)
 def pytest_runtest_call(item):
     if 'trio' in item.keywords:
-        if not iscoroutinefunction(item.obj):
+        print(dir(item.obj))
+        if hasattr(item.obj, 'hypothesis'):
+            # If it's a Hypothesis test, we go in a layer.  We *don't* check
+            # that this is a coroutine function, because we might have wrapped
```

Comment:
I'm concerned about the "might have wrapped", what happened if the test function is synchronous but wrapper in a `hypothesis.given` ? I would say we must always check the test function is async (because we will always call it async still and get a crash if it's not the case).

A simple solution I see is to move the check within _trio_test_runner_factory (given it is currently called just after the check it would change nothing, and it's pretty evident the parameter passed to this factory must be an async function)